cmake/get_info.cmake

```diff
@@ -0,0 +1,310 @@
+    endif()
+    if(CMAKE_BUILD_TYPE MATCHES "MinSizeRel")
+        set(var_string "${var_name}_MINSIZEREL")
+        set(upd_string "#       ${var_name}_MINSIZEREL                  ${${var_string}} ")
+        FILE(APPEND ${log_type} "${upd_string}")
+    endif()
+ENDFUNCTION()
+
+_both(
+"
+
+
+"
+)
+
+IF("${GINKGO_GIT_SHORTREV}" STREQUAL "")
+    _both("
+#########################################################################
+#
+#  Summary of Configuration for  (Ginkgo version ${Ginkgo_VERSION_TAG})\n")
+ELSE()
+    _both("
+#########################################################################
+#
+#  Summary of Configuration for  (Ginkgo version ${Ginkgo_VERSION_TAG}, shortrev ${GINKGO_GIT_SHORTREV})\n")
+ENDIF()
+_both(
+"#
+#
+#  Ginkgo configuration:
+#        CMAKE_BUILD_TYPE:                       ${CMAKE_BUILD_TYPE}
+#        BUILD_SHARED_LIBS:                      ${BUILD_SHARED_LIBS}
+#        CMAKE_INSTALL_PREFIX:                   ${CMAKE_INSTALL_PREFIX}
+#        CMAKE_SOURCE_DIR:                       ${CMAKE_SOURCE_DIR}
+#
+#  User Configuration options:
+#
+#      Enabled modules:
+#        GINKGO_BUILD_OMP:                       ${GINKGO_BUILD_OMP}
+#        GINKGO_BUILD_REFERENCE:                 ${GINKGO_BUILD_REFERENCE}
+#        GINKGO_BUILD_CUDA:                      ${GINKGO_BUILD_CUDA}
+#        GINKGO_BUILD_HIP:                       ${GINKGO_BUILD_HIP}
+#
+#     Tests, benchmarks and examples:
+#        GINKGO_BUILD_TESTS:                     ${GINKGO_DEVEL_TOOLS}
+#        GINKGO_BUILD_EXAMPLES:                  ${GINKGO_BUILD_EXAMPLES}
+#        GINKGO_EXTLIB_EXAMPLE:                  ${GINKGO_EXTLIB_EXAMPLE}
+#        GINKGO_BUILD_BENCHMARKS:                ${GINKGO_BUILD_BENCHMARKS}
+#
+#     Documentation:
+#        GINKGO_BUILD_DOC:                       ${GINKGO_BUILD_DOC}
+#        GINKGO_VERBOSE_LEVEL:                   ${GINKGO_VERBOSE_LEVEL}
+#
+#     Developer helpers:
+#        GINKGO_DEVEL_TOOLS:                     ${GINKGO_DEVEL_TOOLS}
+#        GINKGO_WITH_CLANG_TIDY:                 ${GINKGO_WITH_CLANG_TIDY}
+#        GINKGO_WITH_IWYU:                       ${GINKGO_WITH_IWYU}
+#        GINKGO_CHECK_CIRCULAR_DEPS:             ${GINKGO_CHECK_CIRCULAR_DEPS}
+"
+  )
+_both(
+"#
+#     General information:
+#        CMAKE_BINARY_DIR:                       ${CMAKE_BINARY_DIR}
+#        CMAKE_SOURCE_DIR:                       ${CMAKE_SOURCE_DIR}
+#        CMAKE_CXX_COMPILER:                     ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} on platform ${CMAKE_SYSTEM_NAME} ${CMAKE_SYSTEM_PROCESSOR}
+#                                                ${CMAKE_CXX_COMPILER}"
+  )
+
+
+
+_detailed(
+    "
+#
+#########################################################################
+#
+#   Compiled Modules
+#"
+    )
+_detailed(
+    "
+#########################################################################
+#
+#    The Core module is being compiled.
```

Comment:
suggestion: How about putting this into the `core` directory? Same for the executors

Reply:
Do you mean putting this in a `core.log` and into `${CMAKE_BINARY_DIR}/core/` directory ? I am not sure that there is that much information to put into a separate file.

Reply:
No, I meant more like adding a `component/component_info.cmake` file that outputs component-specific information, the output itself should probably be to a single file.

Reply:
Sounds like a good idea to bring the information management closer to the components/modules themselves in this sort of case.

Reply:
Should be done now.